lrlex/src/lib/parser.rs

```diff
@@ -1,14 +1,31 @@
 use cfgrammar::Span;
-use std::collections::HashMap;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::collections::{HashMap, HashSet};
 use try_from::TryFrom;
 
 use crate::{lexer::Rule, LexBuildError, LexBuildResult, LexErrorKind};
 
 type LexInternalBuildResult<T> = Result<T, LexBuildError>;
+
+lazy_static! {
+    static ref RE_START_STATE_NAME: Regex = Regex::new(r"^[a-zA-Z][a-zA-Z0-9_.]*$").unwrap();
+}
+const INITIAL_START_STATE_NAME: &str = "INITIAL";
```

Comment:
Is it part of the `lex` specification that the initial state is called `INITIAL` or is it normally anonymous or ... ?

Reply:
Yeah, it is typically called initial, the specification at https://pubs.opengroup.org/onlinepubs/007904875/utilities/lex.html says

> The initial state is indicated by the digit '0' or the token INITIAL.

Reply:
So IIUC the above correctly I believe this should allow digits in the first character?

Reply:
Good question, it does look like it should. The easiest thing is probably to try this out on "real" C-lex.

Reply:
My understanding (may have issues) is that a start state is either:
1. A name (so must be a valid c identifier) - I deliberately omitted non-alpha start characters to allow 'special' names at a later date
2. Purely numeric

Thus, the regex would be ^([0-9]+|[a-zA-Z][a-zA-Z0-9_.]*)$
But in the numeric case, I'd prefer to have special handling (so the id is aligned with the parsed result of the input state name)
I feel this could be done as a later enhancement, especially as I've never used numbered start states - they feel much harder to work with (as with any 'magic' numbers)

Reply:
I'm fine with not handling numeric states in this PR (but I'd ask that we raise an issue to remind us that this is a limitation).

Reply:
Raised an issue to track this enhancement: #325

Reply:
Thanks!